### ISSUE TITLE
2.9.1.6 Pre-Alpha: Unable to reclaim orphan calendar entries errors

### ISSUE THREAD

**Issue by Kant-Lavar** [phw198/OutlookGoogleCalendarSync#1326]
> Checklist:-
> * [x] [Upgrade](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Upgrading) to the latest release (alpha if available) ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg)
> * [x] Reproduce the issue
> * [x] Complete the information below
> * [x] Provide your `OGcalsync.log` file (see [how-to](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#locating-your-ogcalsynclog-file))

**OGCS Version**: 2.9.1.6 Pre-Alpha
**Edition**: Installed 
**Sync Direction**: O<->G

**Bug Description**  
Inconsistently getting "Unable to reclaim orphan calendar entries" errors when syncing calendars. Seems to be happening 30-50% of the time, and everything behaves normally otherwise.

**Steps to Reproduce the Issue**  

- Randomly occurs during syncs, commanded or automatic. Unable to isolate any variables indicating if/when issue does occur.

**Any other information**  
Scrubbed through the log file real quick and to my untrained eye it looks like this is a typical error in the log.

```
2021-10-22 11:51:26,545 DEBUG  28 OutlookGoogleCalendarSync.Console [280] -  Reclaiming Outlook calendar entries
2021-10-22 11:51:26,601 DEBUG  28 OutlookGoogleCalendarSync.Console [280] -  Reclaimed: 10/2/2020 8:00 PM (R) "Flyday Friday"
2021-10-22 11:51:26,678 DEBUG  28 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [791] -  Updating the Outlook appointment IDs in Google event.
2021-10-22 11:51:26,684 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  10/2/2020 8:00 PM (R) => "Flyday Friday"
2021-10-22 11:51:26,685 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  OGCSmodified=20200508153213
2021-10-22 11:51:26,685 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  OGCSmodified-01=20211022114943
2021-10-22 11:51:26,685 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_CalendarID=00000000660BD826EEAE934FB6ABA3EF5C8201F80100A1B8EEAF77D6AE479DCEC9FB954C122B0000000002070000
2021-10-22 11:51:26,686 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_CalendarID-01=000000008A7CBC14B33EE24081121FAD3C08860101007588F0243B267A44BBAB93A7AC781F5000000000010D0000
2021-10-22 11:51:26,686 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_EntryID=00000000660BD826EEAE934FB6ABA3EF5C8201F80700A1B8EEAF77D6AE479DCEC9FB954C122B0000000002070000A1B8EEAF77D6AE479DCEC9FB954C122B0000000018090000
2021-10-22 11:51:26,686 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_EntryID-01=000000008A7CBC14B33EE24081121FAD3C08860107007588F0243B267A44BBAB93A7AC781F5000000000010D00007588F0243B267A44BBAB93A7AC781F500000000066950000
2021-10-22 11:51:26,687 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_GlobalApptID=040000008200E00074C5B7101A82E0080000000010ECB8764525D60100000000000000001000000011D8D1759DF0644FAAE48C0816E8345E
2021-10-22 11:51:26,687 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_GlobalApptID-01=040000008200E00074C5B7101A82E008000000002068F58D3AC7D701000000000000000010000000C14316E6EA40FF44AC0ABB6822AC1452
2021-10-22 11:51:26,687 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [876] -  Saving timestamp when OGCS updated event.
2021-10-22 11:51:26,790 FAIL  28 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [1847] -  Google.Apis.Requests.RequestError
Not Found [404]
Errors [
	Message[Not Found] Location[ - ] Reason[notFound] Domain[global]
]

2021-10-22 11:51:26,791 WARN  28 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [1904] -  Unhandled API exception.
2021-10-22 11:51:26,792 DEBUG  28 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [898] -  Attempted to update event in calendar: kant.lavar@gmail.com
2021-10-22 11:51:26,793 WARN  28 OutlookGoogleCalendarSync.Console [274] -  Failure processing Outlook item:-
2021-10-22 11:51:26,794 WARN  28 OutlookGoogleCalendarSync.Console [274] -  10/2/2020 8:00 PM (R) "Flyday Friday"
2021-10-22 11:51:26,818 ERROR  28 OutlookGoogleCalendarSync.Console [278] -  Unable to reclaim orphan calendar entries in Outlook calendar.
2021-10-22 11:51:27,105 ERROR  28 OutlookGoogleCalendarSync.OGCSexception [28] -  Google.GoogleApiException: Google.Apis.Requests.RequestError
Not Found [404]
Errors [
	Message[Not Found] Location[ - ] Reason[notFound] Domain[global]
]

2021-10-22 11:51:27,106 ERROR  28 OutlookGoogleCalendarSync.OGCSexception [21] -  Code: 0x80131500,-2146233088; Location: MoveNext() in <Unknown File>
2021-10-22 11:51:27,106 ERROR  28 OutlookGoogleCalendarSync.OGCSexception [65] -     at Google.Apis.Requests.ClientServiceRequest`1.<ParseResponse>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Google.Apis.Requests.ClientServiceRequest`1.Execute()
   at OutlookGoogleCalendarSync.GoogleOgcs.Calendar.UpdateCalendarEntry_save(Event& ev) in C:\Users\Paul\Git\OutlookGoogleCalendarSync\src\OutlookGoogleCalendarSync\GoogleOgcs\GoogleCalendar.cs:line 898
   at OutlookGoogleCalendarSync.OutlookOgcs.Calendar.ReclaimOrphanCalendarEntries(List`1& oAppointments, List`1& gEvents) in C:\Users\Paul\Git\OutlookGoogleCalendarSync\src\OutlookGoogleCalendarSync\OutlookOgcs\OutlookCalendar.cs:line 832
   at OutlookGoogleCalendarSync.Sync.Engine.Calendar.synchronize() in C:\Users\Paul\Git\OutlookGoogleCalendarSync\src\OutlookGoogleCalendarSync\Sync\Calendar.cs:line 442
   at OutlookGoogleCalendarSync.Sync.Engine.Calendar.<>c__DisplayClass11_1.<StartSync>b__0(Object o, DoWorkEventArgs args) in C:\Users\Paul\Git\OutlookGoogleCalendarSync\src\OutlookGoogleCalendarSync\Sync\Calendar.cs:line 124
```

[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/7400529/OGcalsync.log)

**Comment by Kant-Lavar** [comment]
Don't have a log file from it yet. but it looks like the same thing is happening on a second computer I tried to set this up on as well. Will try and get a log from that second computer uploaded as soon as I can.

**Comment by phw198** [comment]
Thanks for reporting and your log files @Kant-Lavar 

I've just posted what I'm hoping is the fix to this over on #1314 with v2.9.1.11 if you would like to give that a go.
